doc/debugging.rst

```diff
@@ -14,33 +14,91 @@ This allows breakpoints and the execution flow to be viewed and controlled right
 
-* Multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+.. _console commands: https://pythonconquerstheuniverse.wordpress.com/2009/09/10/debugging-in-python/
 
-  * By selecting the respective option from the Debug menu.
-  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
-  * By double-clicking to the left of the line number in an open file.
-  * With an ``ipdb.set_trace()`` statement in your code (after ``import`` ing ``pdb``).
-  * Interactively, using the ``b`` command in an ``ipdb`` session.
+GIF: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open (commands from the toolbar)
+
+Spyder's debugger offers syntax highlighting, code completion and command history, which work exactly like they do in the normal interactive interpreter.
+Use the up and down arrows to recall previous commands, and press :kbd:`Tab` to trigger autocomplete suggestions.
+
+GIF: Browsing autocomplete suggestions
+
+Furthermore, IPython's `magic functions`_ are available in debugging mode.
+You can, for example, run ``%matplotlib <BACKEND>`` to switch between the ``qt5`` and ``inline`` plotting backends, or ``%timeit`` to check how fast a given snippet of code is.
+
+.. _magic functions: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+
+SCREENSHOT: (timeit)
+
+SCREENSHOT of highlighting step (close up)
+
+Finally, you can enter and execute multiline statements in Spyder's debugger just like with the regular IPython prompt, to easily execute complex code while debugging.
+
+GIF of multiline debugging
+
+
+
+===========
+Breakpoints
+===========
+
+Spyder's debugger is integrated with the :guilabel:`Breakpoints` pane, which lists the file, line, and condition (if any) of every breakpoint defined.
+To open it, select :menuselection:`Debug --> List breakpoints`, or press :kbd:`Ctrl-Shift-B` by default.
 
 .. image:: images/debugging/breakpoints-standard.png
    :alt: Spyder's Breakpoints panel, with a number of examples showing file, line number and an optional condition
 
-* A :guilabel:`Breakpoints` pane, listing the file, line, and condition (if any) of every breakpoint defined (:menuselection:`&Debug --> List breakpoints`, or :kbd:`Ctrl-Shift-B` by default).
+There are several different ways to set and clear breakpoints:
+
+  * With the :guilabel:`Set/clear breakpoint` option in the Debug menu.
+  * Through pressing the configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
+  * By clicking to the left of the line number in an open file in the Editor (adding :kbd:`Shift` for a conditional breakpoint).
+  * With the ``breakpoint()`` function in your code.
+  * Interactively, using the ``!b`` command in an ``ipdb`` session.
```

Comment:
```suggestion
  * Interactively, using the ``b`` command in an debugging session.
```

* This comes before the explanation about using `!` for Pdb commands, so I think we should mention the command all people know.
* `ipdb session` sounds too technical to me.

Reply:
> This comes before the explanation about using ! for Pdb commands, so I think we should mention the command all people know.

I thought we already talked about this and decided on ``!b``, since if users have a variable with the same name (not unlikely, especially for beginners who are unlikely to know the ipdb commands anyway or know what to do when something goes wrong), it won't work. And if "all users" already know the command (which I rather doubt), they don't need to be told it.

Also, fixed grammar error resulting from the ("an" -> "an"

```suggestion
  * Interactively, using the ``!b`` command in a debugging session.
```

Reply:
I really disagree with using `!b` here. It'd be confusing for people that know Pdb commands (it'd look to them as a typo).

If people type `b` and don't see their breakpoints, they'll get the informative message we added telling them what to do. So I think it's better to leave the usage of `!` to its own section below.

Reply:
Okay, I guess that makes sense. Therefore, with that change, the grammar-corrected revision,

```suggestion
  * Interactively, using the ``b`` command in a debugging session.
```